kernel/packages/unity-interface/BrowserInterface.ts

```diff
@@ -377,6 +378,45 @@ export class BrowserInterface {
     globalThis.globalStore.dispatch(updateFriendship(action, userId.toLowerCase(), false))
   }
 
+  public SearchENSOwner(data: { name: string; maxResults: number }) {
+    let profilesPromise
+
+    if (/^0x[a-fA-F0-9]{40}$/.test(data.name)) {
+      profilesPromise = ProfileAsPromise(data.name, undefined, ProfileType.DEPLOYED).then((profile) => {
+        return [profile]
+      })
+    } else {
+      profilesPromise = new Promise((resolve, reject) => {
```

Comment:
```suggestion
      profilesPromise = new Promise(async (resolve, reject) => {
```
I think that if you do this you can avoid the anonymous function below

Reply:
Another approach to this, is to do 

```ts

async function getProfiles() {

// your code with awaits and throws

}



profilesPromise = getProfiles()


```

that would make the code easier to maintain also

Reply:
moved the whole logic to it own file `fetchENSOwnerProfile.ts` to make it easier to mantain